core/src/test/java/org/apache/iceberg/TestTableMetadata.java

```diff
@@ -232,8 +231,6 @@ public void testJsonConversion() throws Exception {
     assertThat(metadata.statisticsFiles()).isEqualTo(statisticsFiles);
     assertThat(metadata.partitionStatisticsFiles()).isEqualTo(partitionStatisticsFiles);
     assertThat(metadata.refs()).isEqualTo(refs);
-    assertThat(metadata.rowLineageEnabled()).isEqualTo(expected.rowLineageEnabled());
-    assertThat(metadata.nextRowId()).isEqualTo(expected.nextRowId());
```

Comment:
Looks like this should probably be restored as well. I'll fix it in a follow up.

Reply:
Yeah these just need a slight modification

Reply:
Yeah I think the `nextRowId` assertions should still be retained